app/src/main/java/org/thoughtcrime/securesms/preferences/SettingsActivity.kt

```diff
@@ -149,12 +138,14 @@ class SettingsActivity : PassphraseRequiredActionBarActivity() {
         if (result.resultCode != Activity.RESULT_OK) return@registerForActivityResult
 
         val outputFile = Uri.fromFile(File(cacheDir, "cropped"))
-        val inputFile: Uri? = result.data?.data ?: tempFile?.let(Uri::fromFile)
+        val inputFile: Uri? = result.data?.data ?: viewModel.getTempFile()?.let(Uri::fromFile)
```

Comment:
While unlikely, `inputFile` could be null here because `createTempFile` can fail - so perhaps log something rather than proceed to `cropImage` with a null file if `inputFile` is null.

Reply:
Yeah, this follows the logic that was currently in place. I'm not exactly sure how it can reach a state where the VM's temp file is null. We could maybe make sure it never is. But the input file here already expects the Uri to be nullable